src/record_syscall.cc

```diff
@@ -6006,6 +6006,44 @@ static vector<WriteHole> find_holes(RecordTask* t, int desc, uint64_t offset, ui
   return ret;
 }
 
+static void check_outside_mappings(const string& tracee_file_name, const RecordSession& session) {
+  DIR* proc = opendir("/proc");
+  if (!proc) {
+    FATAL() << "Failed to read /proc";
+  }
+  while (true) {
+    struct dirent* f = readdir(proc);
+    if (!f) {
+      break;
+    }
+    pid_t pid = atoi(f->d_name);
+    if (pid) {
+      bool ok = true;
+      for (auto it : session.tasks()) {
+        if (pid == it.second->tid) {
+          ok = false;
+          break;
+        }
+      }
+      if (!ok) {
+        break;
```

Comment:
I'm confused ... it looks like we stop looping over all processes as soon as we find one process that we're recording. Shouldn't this be `continue`?